File: tests/fixtures/php/enums.php
<?php

enum Status {
    case Active;
    case Inactive;
    case Pending;
}

enum Color: string {
    case Red = 'red';
    case Blue = 'blue';
    case Green = 'green';

    public function label(): string {
        return ucfirst($this->value);
    }
}

enum Priority: int implements Cacheable {
    case Low = 1;
    case Medium = 5;
    case High = 10;

    public function cacheKey(): string {
        return 'priority_' . $this->value;
    }

    public function cacheTTL(): int {
        return 3600;
    }
}

File: tests/fixtures/java/enums.java
package com.example.models;

public enum Status {
    ACTIVE,
    INACTIVE,
    PENDING;

    public String label() {
        return name().toLowerCase();
    }
}

public enum HttpMethod {
    GET("GET"),
    POST("POST"),
    PUT("PUT"),
    DELETE("DELETE");

    private final String value;

    HttpMethod(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }
}

public enum Color {
    RED,
    GREEN,
    BLUE
}
